### ISSUE TITLE
Add more docs examples for offset encoding

### ISSUE THREAD

**Issue by joelostblom** [vega/altair#2604]
It would be nice if we could add a page about the new offset encoding that contained common examples. In addition to [the offset barplot in the VegaLite docs](https://vega.github.io/vega-lite/docs/encoding.html#positon-offset), I think these examples would be nice:

```python
import altair as alt
from vega_datasets import data

source = data.barley.url

alt.Chart(source).mark_point().encode(
    x='site:O',
    y='yield:Q',
    xOffset='year:N',
    color='year:N',
)
```

![image](https://user-images.githubusercontent.com/4560057/168725016-bab20591-3575-4836-970d-32a84e4402de.png)

```python
alt.Chart(source).mark_point().encode(
    x='site:O',
    y='yield:Q',
    xOffset=alt.XOffset('offset:Q', scale=alt.Scale(domain=(0, 3)))
).transform_calculate(
    offset='random()'
)
```
![image](https://user-images.githubusercontent.com/4560057/168725049-81496cce-8431-472a-864d-d770699491ab.png)

```python
alt.Chart(source, width=alt.Step(40)).mark_point().encode(
    x='site:O',
    y='yield:Q',
    xOffset=alt.XOffset('offset:Q', scale=alt.Scale(domain=(0, 10))),
    color='year:N'
).transform_calculate(
    offset="datum.year == 1932 ? 5 + random() : 0 + random()"
)
```

![image](https://user-images.githubusercontent.com/4560057/168725093-6ee0d928-4228-40cf-afde-b1b9a1a5b9c0.png)

**Comment by joelostblom** [comment]
I don't think we have anything planned currently for the encoding section so please feel free to start a PR on that. In case you want to work on the encoding section and have time it might make sense that you take over work on https://github.com/altair-viz/altair/pull/2607 as well to get familiar with the format of those doc sections? I left some comments there that needs to be addressed but other than that I think it is close to a merge and I am happy to review as soon as possible after you make changes. Any of the things we have discussed that you are interested in and have time for would be a welcome contribution, just let me know what you prefer when you start working on it so that we don't work onthe same thing separately (although I will likely not get to working on that PR until after Christmas myself).

**Comment by binste** [comment]
Just so you are aware, I will work on the encoding pages next, expanding the current one and potentially adding separate ones for certain concepts similar to the vega-lite docs. I will also clarify the distinction between `value` and `datum` as discussed in #2572

**Comment by mattijn** [comment]
Just to check @binste, I could not find the documentation in the encoding channel options on x/yOffset in https://binste.github.io/altair-docs/user_guide/encodings/channel_options.html. 

I think it be as included as follow:

```cmd
XOffset and YOffset
~~~~~~~~~~~~~~~~~~~

 The :class:`XOffset` and :class:`YOffset` encodings accept the following options:

 .. altair-object-table:: OffsetDef
```

**Comment by dangotbanned** [comment]
Closing as completed

These examples are referenced in https://altair-viz.github.io/user_guide/encodings/channels.html#position
- https://altair-viz.github.io/gallery/grouped_bar_chart2.html#gallery-grouped-bar-chart2
- https://altair-viz.github.io/gallery/strip_plot_jitter.html#gallery-strip-plot-jitter
